src/index.ts

```diff
@@ -1,123 +1,106 @@
+		message: "Which ESLint configuration do you want to install?",
+		choices: availableConfigs.map((config) => ({
+			title: config,
+			value: config
+		}))
+	}).then(({config}) => config)
 }
 
 function createESLintConfig(whichConfig: AvailableConfig) {
-	const { config } = configs[whichConfig]
-
 	const dataAsString = JSON.stringify(
-		config,
+		{
+      extends: `@monogram/eslint-config/${whichConfig}`
+    },
 		null,
-		2 // TODO: get config from prettier
+		2 
 	)
 
 	writeFileSync(`${process.cwd()}/${ESLINT_FILENAME}`, `module.exports = ${dataAsString}`)
 
 	console.log(`✅ ESLint configuration file created at ${process.cwd()}/${ESLINT_FILENAME}`)
 }
 
-async function choosePackageManager() {
-	let packageManager = findPackageManager()
-
-	if (packageManager) {
-		console.log(`\n📦 An existing ${packageManager} installation was found`)
-	} else {
-		const packageManagersOptions = packageManagers.join(', ')
-
-		const chosenManager = await question(
-			`Which package manager should be used? \n[${packageManagersOptions}] `
-		)
-
-		if (packageManagersOptions.includes(chosenManager as PackageManager)) {
-			packageManager = chosenManager as PackageManager
-		}
-	}
-
-	return packageManager
+async function choosePackageManager(): Promise<PackageManager> {
+	return prompts({
+		type: 'select',
+		name: 'packageManager',
+		message: "Which ESLint configuration do you want to install?",
+		choices: packageManagers.map((pm) => ({
+			title: pm,
+			value: pm
+		})),
+		initial: findPackageManager()
+	}).then(({packageManager}) => packageManager)
 }
 
-async function installDependencies(config: AvailableConfig, packageManager: PackageManager) {
-	const installPrefix = installPrefixes[packageManager]
+const installPrefixes = {
+	yarn: 'yarn add -D',
+	pnpm: 'pnpm add -D',
+	npm: 'npm i -D'
+} as const
 
-	const { dependencies } = configs[config]
-	const dependenciesAsString = dependencies.join(' ')
+async function installDependency(packageManager: PackageManager) {
+	const installPrefix = installPrefixes[packageManager]
 
-	const installCommand = `${installPrefix} ${dependenciesAsString}`
+	const installCommand = `${installPrefix} eslint @monogram/eslint-config`
 
 	console.log(`📦 Installing dependencies...`)
-	console.log(`${installCommand}\n`)
-
-	await exec(
-		installCommand,
-		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-		// @ts-ignore
-		(stdout: string, stderr: string) => {
-			if (stderr) {
-				console.error(stderr)
-			} else {
-				console.log(stdout)
-			}
-
-			Promise.resolve({ stdout, stderr })
-		}
-	)
+
+  try {
+    execSync(installCommand)
+  } catch (error) {
+    console.error(error)
+  }
 }
+
+const lockFiles = {
+	yarn: `${process.cwd()}/yarn.lock`,
+	pnpm: `${process.cwd()}/pnpm-lock.yaml`,
+	npm: `${process.cwd()}/package-lock.json`
+}
+
+function findPackageManager() {
+	const index = packageManagers.findIndex((pm) => existsSync(lockFiles[pm]))
+	if (index === -1) return null
+	return index
+}
```

Comment:
I don't think this needed a refactor? Please keep the previous version.

Reply:
The reason I refactored this is because I need the index of the `packageManager`, because instead of skipping the "choose package manager" step I now show it every time, but the default option highlighted is their current package manager (so they can just hit enter).

I can technically do it without getting the index from the function but then i would need to do `packageManagers[packageManager]` to find the index on the receiving end, because `prompts` library expects an index for the "default" option.

Reply:
Gotcha, let's see how it goes with inquirer. I'd recommend keeping the function `findPackageManager` more or less as it was before, returning `null` or the package manager string, then as this was specific for prompts we could get the index only when it's used(like `packageManagers.indexOf(findPackageManager() || DEFAULT_PACKAGE_MANAGER)`